operator/v1alpha1/types_etcdbackup.go

```diff
@@ -0,0 +1,102 @@
+}
+
+type EtcdBackupSpec struct {
+	// PVCName specifies the name of the PersistentVolumeClaim which binds a PersistentVolume where the
+	// etcd backup file would be saved
+	// +kubebuilder:validation:Required
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="pvcName is immutable once set"
+	PVCName string `json:"pvcName"`
+}
+
+const (
+	// Etcd backup is running
+	Running EtcdBackupState = "Running"
+	// Etcd backup is completed
+	Completed EtcdBackupState = "Completed"
+	// Etcd backup failed
+	Failed EtcdBackupState = "Failed"
+	// Etcd backup is pending
+	Pending EtcdBackupState = "Pending"
+)
+
+// etcdBackupState declares valid gathering state types
+// +kubebuilder:validation:Optional
+// +kubebuilder:validation:Enum=Running;Completed;Failed;Pending
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Running' && self == 'Pending')", message="etcdBackupState cannot transition from Running to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Completed' && self == 'Pending')", message="etcdBackupState cannot transition from Completed to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Failed' && self == 'Pending')", message="etcdBackupState cannot transition from Failed to Pending"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Completed' && self == 'Running')", message="etcdBackupState cannot transition from Completed to Running"
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Failed' && self == 'Running')", message="etcdBackupState cannot transition from Failed to Running"
+type EtcdBackupState string
+
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.startTime) || has(self.startTime))",message="cannot remove startTime attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.finishTime) || has(self.finishTime))",message="cannot remove finishTime attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.etcdBackupState) || has(self.etcdBackupState))",message="cannot remove etcdBackupState attribute from status"
+// +kubebuilder:validation:Optional
+type EtcdBackupStatus struct {
+	// conditions provide details on the status of the gatherer job.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []metav1.Condition `json:"conditions" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// etcdBackupState reflects the current state of the etcd backup process.
+	// +kubebuilder:validation:Optional
+	// +optional
+	State EtcdBackupState `json:"etcdBackupState,omitempty"`
+
+	// startTime is the time when the etcd backup started.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="startTime is immutable once set"
+	// +optional
+	StartTime metav1.Time `json:"startTime,omitempty"`
```

Comment:
pointer

Reply:
I would entirely remove StartTime and FinishTime. It's possible to reconcile this from the job status conditions, but is there a reason we need this?

Reply:
The start and finish times aren't strictly necessary. This was just my attempt to provide more context in the status, but on second thought this info can be gathered from the condition LastTransitionTime as well.
Will remove them in that case.